hemera_udf/aave_v2/export_aave_v2_job.py

```diff
@@ -0,0 +1,408 @@
+                    if vary_token in address_token_block_balance_dic[address]:
+                        vary_debt = address_token_block_balance_dic[address][vary_token][block_number]
+                stable_token = debt_reserve.stable_debt_token_address
+                stable_debt = 0
+                if address in address_token_block_balance_dic:
+                    if stable_token in address_token_block_balance_dic[address]:
+                        stable_debt = address_token_block_balance_dic[address][stable_token][block_number]
+                a_record.debt_after_liquidation = vary_debt + stable_debt
+
+                collateral_asset = a_record.collateral_asset
+                collateral_reserve = self.reserve_dic[collateral_asset]
+
+                a_record.collateral_after_liquidation = address_token_block_balance_dic[address][
+                    collateral_reserve.a_token_address
+                ][block_number]
+
+                res_d[address][collateral_asset].asset = collateral_asset
+                res_d[address][collateral_asset].address = address
+                res_d[address][collateral_asset].supply_amount = a_record.collateral_after_liquidation
+                res_d[address][collateral_asset].block_number = a_record.block_number
+                res_d[address][collateral_asset].block_timestamp = a_record.block_timestamp
+
+                res_d[address][debt_asset].asset = debt_asset
+                res_d[address][debt_asset].address = address
+                res_d[address][debt_asset].borrow_amount = a_record.debt_after_liquidation
+                res_d[address][debt_asset].block_number = a_record.block_number
+                res_d[address][debt_asset].block_timestamp = a_record.block_timestamp
+
+                # record last liquidation time and amount
+                self._collect_item(
+                    AaveV2LiquidationAddressCurrentD.type(),
+                    AaveV2LiquidationAddressCurrentD(
+                        address=address,
+                        asset=collateral_asset,
+                        last_liquidation_time=a_record.block_timestamp,
+                        last_total_value_of_liquidation=a_record.liquidated_collateral_amount,
+                        block_number=a_record.block_number,
+                    ),
+                )
+        address_currents = []
+        for address, outer_dic in res_d.items():
+            for reserve, kad in outer_dic.items():
+                address_currents.append(kad)
+        self._collect_items(AaveV2AddressCurrentD.type(), address_currents)
+        self._merge_dataclasses(AaveV2ReserveDataD, ["asset"])
+        self._merge_dataclasses(AaveV2ReserveDataCurrentD, ["asset"])
+        self._merge_dataclasses(AaveV2LiquidationAddressCurrentD, ["address", "asset"])
+
+        logger.info("This batch of data have processed")
+
+    def _merge_dataclasses(self, data_class, attributes):
```

Comment:
This can be a util function. Or we might already have it.

Reply:
I don't see one. added it to utils/collection_util.py